pkg/cmd/provisioning/identity_provider.go

```diff
@@ -0,0 +1,335 @@
+package provisioning
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	jose "gopkg.in/square/go-jose.v2"
+
+	"github.com/openshift/cloud-credential-operator/pkg/aws"
+)
+
+var (
+	// IdentityProviderOpts captures the options that affect creation of the identity provider
+	IdentityProviderOpts = options{
+		InfraName:     "",
+		Region:        "",
+		PublicKeyPath: "",
+	}
+
+	// discoveryDocumentTemplate is a template of the discovery document that needs to be populated with appropriate values
+	discoveryDocumentTemplate = `{
+	"issuer": "%s",
+	"jwks_uri": "%s/%s",
+    "response_types_supported": [
+        "id_token"
+    ],
+    "subject_types_supported": [
+        "public"
+    ],
+    "id_token_signing_alg_values_supported": [
+        "RS256"
+    ],
+    "claims_supported": [
+        "aud",
+        "exp",
+        "sub",
+        "iat",
+        "iss",
+        "sub"
+    ]
+}`
+)
+
+type JSONWebKeySet struct {
+	Keys []jose.JSONWebKey `json:"keys"`
+}
+
+func createIdentityProvider(client aws.Client, infraName, region, publicKeyPath, targetDir string) error {
+	bucketName := fmt.Sprintf("%s-installer", infraName)
+	issuerURL := fmt.Sprintf("https://%s.s3.%s.amazonaws.com", bucketName, region)
+
+	var bucketTagValue string
+	_, err := client.CreateBucket(&s3.CreateBucketInput{
```

Comment:
For the `--dry-run` param (or whatever we end up calling it), we need to also provide the JSON (or YAML) of the AWS object that needs to be created/updated. Something like this (I believe you can leave out parameters that you want as defaults):
```
[jdiaz@minigoomba ~]$ aws s3api create-bucket --generate-cli-skeleton
{
    "ACL": "authenticated-read",
    "Bucket": "",
    "CreateBucketConfiguration": {
        "LocationConstraint": "ap-east-1"
    },
    "GrantFullControl": "",
    "GrantRead": "",
    "GrantReadACP": "",
    "GrantWrite": "",
    "GrantWriteACP": "",
    "ObjectLockEnabledForBucket": true
}

```